Added a helm chart

ineentho opened the pull request:
I've created a helm chart for deploying renovate-pro + postgresql, offering a similar experience to the docker-compose files except for Kubernetes.

I can't say that I've tested everything completely, but I'm using most of the features the chart has to offer (.npmrc, config.js, the postgresql subchart, ingress with https) on my Gitlab CE install and everything seems to work just fine this far.

For configuring the renovate core, I took the approach of inlining the config.js inside of values.yaml because I think that's most the easiest to understand (a clear separation between core and pro config), and what will offer most power.

I didn't include any documentation, but I'd be happy to add that if you'd like me to, but before we add that the values.yaml should be sufficent documentations for anybody with some helm experience.

I'd recommend hosting this chart somewhere if you'd like to match Gitlabs helm chart as you mentioned in #33. The easiest way to publish this chart is probably something like:
```
mkdir $chart-repo
helm package $pro/helm-charts/renovate-pro -d $chart-repo
helm repo index $chart-repo
# publish $chart-repo using a static file server
```

For anybody who wants to try this chart out right now: Make a copy of values.yaml and store it somewhere outside of this repo, and run `helm upgrade renovate-pro --install -f values.yaml $pro/helm-charts/renovate-pro`

gabemeola commented on helm-charts/renovate-pro/values.yaml:
```diff
@@ -0,0 +1,122 @@
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+replicaCount: 1
```
Curious, will Renovate work with more than one replica?

rarkins replied:
No

cobexer commented:
Please push the Helm Chart to the stable repository https://github.com/helm/charts/blob/master/CONTRIBUTING.md instead of adding the chart here when you get around to it.

ineentho commented:
I actually think it should live here or alternatively at something like renovatebot/pro-charts. Helm is moving away from the central repositories and v3 doesn't even include stable by default.

@rarkins Are you still interested in managing this helm chart? If so, I'm willing to take another look at this PR and updating and cleaning it up a bit.

aslafy-z commented:
Hello @ineentho, I added some fields and bumped version of Renovate in this PR: https://github.com/ineentho/pro/pull/1

etiennetremel commented:
Is there anything preventing this PR to be merged?

rarkins commented:
@etiennetremel if you have tested it and can also verify that it works for you then I'd be happy to merge it. Or if anyone else for that matter!

aslafy-z commented:
I tested it and I confirm there are things to change. @ineentho see my PR linked above, please update yours or I will create a new one.

ineentho commented:
Thanks @aslafy-z, I've merged your changes and added you as a contributor if you want to make any further changes.

Also, I removed the replica count option as per gabemeolas comment above.

I'm not currently using the self hosted version of renovate, so I cannot verify that the changes will work. @aslafy-z, feel free to take over from here to get it merged.

gabemeola commented:
We've been using this is production for about 6 months running against a self hosted Gitlab. Other than the `renovate/pro` image not as up to date as the open source offering, it been working great!

aslafy-z commented:
@rarkins Should I rename the chart to `WhiteSource Renovate On-Premise`?

rarkins commented:
> @rarkins Should I rename the chart to `WhiteSource Renovate On-Premises`? or without `On-Premises`?

I think "WhiteSource Renovate" is fine. We just use the On-Premises to clarify between the hosted app and this self-hosted image when necessary, so in this case it shouldn't be necessary.

aslafy-z commented:
This link to the Renovate User Agreement is dead: https://renovatebot.com/user-agreement
Should I replace it with another one?

rarkins commented:
@aslafy-z thanks: https://www.whitesourcesoftware.com/terms-of-service/

aslafy-z commented:
I think it's pretty good for now, some more enhancements can be added via subsequent PRs.
I've listed some environment variables I found that could be added as config values, much of them being not documented I did not implemented them.

- PIP_INDEX_URL  
- RENOVATE_CACHE_NPM_MINUTES
- YARN_MUTEX_FILE
- RENOVATE_SKIP_CACHE
- FORCE_COLOR
- RENOVATE_REUSE_PACKAGE_LOCK
- RENOVATE_PAGINATE_ALL
- GITLAB_IGNORE_REPO_URL

Also, adding support for custom cache volume would be great.

etiennetremel commented:
For hosting the helm repo, I would suggest using [chart releaser](https://github.com/helm/chart-releaser). When used together with the Github page feature you can turn it into a Helm registry.

See an example of implementation: https://github.com/helm/charts-repo-actions-demo

rarkins commented:
Needs a little updating before merge. e.g. the postgres container is no longer used.

aslafy-z commented:
I made the requested changes. @rarkins can you review?

This chart can be tested without packaging using [`helm-git`](https://github.com/aslafy-z/helm-git)

```shell
helm plugin install https://github.com/aslafy-z/helm-git
helm repo add whitesource-renovate 'git+https://github.com/ineentho/pro@helm-charts/whitesource-renovate?ref=helm-chart'
wget https://raw.githubusercontent.com/ineentho/pro/helm-chart/helm-charts/whitesource-renovate/values.yaml
# Edit values.yaml accordingly to your setup
helm upgrade --install whitesource-renovate -f values.yaml whitesource-renovate/whitesource-renovate
```

rarkins commented on helm-charts/whitesource-renovate/values.yaml:
```diff
@@ -0,0 +1,92 @@
+  # API endpoint, include trailing slash. Examples: https://api.github.com/, https://gitlab.com/api/v4/
+  renovateEndpoint:
+
+  # Personal Access Token for bot account
```
```suggestion
  # Personal Access Token for bot account (GitLab only)
```

rarkins commented on helm-charts/whitesource-renovate/values.yaml:
```diff
@@ -0,0 +1,92 @@
+  # Personal Access Token for bot account
+  renovateToken:
+
+  #  The GitHub App ID provided by GitHub Enterprise when you provisioned the WhiteSource Renovate app. Force string format by quotting value.
```
```suggestion
  #  The GitHub App ID provided when you provisioned the WhiteSource Renovate app. Force string format by quoting value.
```

aslafy-z commented:
@rarkins I made the requested changes. FYI, I was not from now able to test this later version

rarkins commented:
Is there anyone who could test/validate this chart before we merge it?

baurmatt commented on helm-charts/whitesource-renovate/templates/deployment.yaml:
```diff
@@ -0,0 +1,155 @@
+            - name: LICENSE_KEY
+              valueFrom:
+                secretKeyRef:
+                  secretName: {{ .Release.Name }}
```
I think for current version of k8s this should be `name`. At least this fails on our `v1.17.6` cluster. There are a couple more places where this needs to be replaced.

aslafy-z replied:
Confirmed in the doc, it should be `name` when using `secretKeyRef`. Updated the manifests to fix that!

baurmatt commented on helm-charts/whitesource-renovate/templates/deployment.yaml:
```diff
@@ -0,0 +1,155 @@
+        - name: {{ .Release.Name }}-npmrc-volume
+          secret:
+            secretName: {{ .Release.Name }}-npmrc
+        {{- end }
```
```suggestion
        {{- end }}
```

aslafy-z replied:
Nice catch!

baurmatt commented on helm-charts/whitesource-renovate/templates/deployment.yaml:
```diff
@@ -0,0 +1,155 @@
+              subPath: .npmrc
+            {{- end }}
+            - name: {{ .Release.Name }}-cache-volume
+              mountPath: /tmp
```
```suggestion
              mountPath: /tmp/renovate
```

In an earlier version this already was `/tmp/renovate`. I think this made sense because putting `/tmp` on the volume would also put system temp files on it.

aslafy-z replied:
Ok for me but Docker compose files will have to be updated too

baurmatt commented:
I've added one last suggestion for the `renovate-cache-volume` mount path. Besides this, everything looks good and I would deploy it straight to production ;)

rarkins commented:
Thanks everyone!
